<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Interpreting the POUMM • POUMM</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">POUMM</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/UserGuide.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/InterpretingPOUMM.html">Interpreting the POUMM</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Interpreting the POUMM</h1>
                        <h4 class="author">Venelin Mitov</h4>
            
          </div>

    
    
<div class="contents">
<p>Here, we consider the POUMM from the point of view of its biological interpretation. Before we begin, we need to make an important cautionary note:</p>
<p>One should always be aware that non-mechanistic stochastic models like Brownian motion (BM) and Ornstein-Uhlenbeck (OU) are unlikeliy to be accurate representations of the trait evolution at the individual level, be it an individual taxon, a patient infected with a pathogen, or any other evolving entity represented by a lineage in a phylogeny. Rather, these models are useful when analyzing population properties, such as the mean trait value, variance, covariance and correlation. In terms of the models, these properties represent deterministic functions of the model parameters and the temporal information in the tree (e.g. mean root-tip distance and phylogenetic distance between couples of tips). This makes it possible to use a statistical fit of the model parameters to the tree and the data (e.g. maximum likelihood and posterior samples), to estimate point-values and credibility intervals of such population properties. All these estimates depart from the assumption that the model is at least approximately accurate at the population level. If the data violates this assumption, the resulting estimates and credibility intervals can be strongly biased. For an example of such bias caused by a wrong Brownian motion assumption, we refer the reader to <span class="citation">(Mitov and Stadler 2016)</span>. Therefore, we recommend comparing model-based parametric estimates with non-parametric estimates, such as sample mean, variance and covariance for sub-populations (e.g. stratifications) of the data <span class="citation">(Mitov and Stadler 2016)</span>.</p>
<div id="parametrizing-the-poumm" class="section level1">
<h1 class="hasAnchor">
<a href="#parametrizing-the-poumm" class="anchor"></a>Parametrizing the POUMM</h1>
<p>From a modeling perspective, the POUMM can be regarded as a combination of two of its widely used nested models:</p>
<ul>
<li>The non-mixed phylogenetic Ornstein-Uhlenbeck (POU) model, (also referred to as the ‘Hansen model’ <span class="citation">(Butler and King 2004)</span>), which has been used to infer local phenotypic optima in phylogenetic clades or across categories of phylogenetic related species <span class="citation">(Butler and King 2004)</span>, and to prove the presence of stabilizing selection towards a single stationary peak (SSP) <span class="citation">(Harmon et al. 2010)</span>.</li>
<li>The phylogenetic mixed model (PMM) <span class="citation">(Lynch 1991; Housworth, Martins, and Lynch 2004)</span>, which has been used to measure phylogenetic heritability under the assumption that the genotype evolves according to Brownian motion <span class="citation">(Hodcroft et al. 2014)</span>;</li>
</ul>
<p>The POUMM combines the applications of the above two models. The properties we will consider represent bijective functions of some of the POUMM parameters. Thus, it is possible to reparametrize the POUMM, so that the model inference is done directly on properties of interest, e.g. the phylogenetic heritability. This is particularly useful for Bayesian inference, since for Bayesian inference priors should be specified for the properties of interest rather than the default POUMM parameters. We call a <em>parametrization</em> any numerical bijective function mapping its argument into the default POUMM parameter-space (<span class="math inline">\(&lt;g_0,\alpha,\theta,\sigma,\sigma_e&gt;\)</span>).</p>
<div id="trait-distribution-at-equilibrium" class="section level2">
<h2 class="hasAnchor">
<a href="#trait-distribution-at-equilibrium" class="anchor"></a>Trait distribution at equilibrium</h2>
<p>An interesting property of the POUMM is that, in the limit <span class="math inline">\(t\to\infty\)</span>, it defines a stationary normal distribution for the heritable component (<span class="math inline">\(g\)</span>) at the tips with mean <span class="math inline">\(\theta\)</span> and a variance-covariance matrix:</p>
<span class="math display">\[\begin{multline}\label{eq:SigmaOUEquilibrium}
\begin{array}{lll}
\Sigma_{ii}  = \sigma_{\infty}^2 &amp; = &amp; \frac{\sigma^2}{2\alpha} \\
\Sigma_{ij} =  \Sigma_{ij,\infty} &amp; = &amp; \sigma_{\infty}^2\,e^{-\alpha{d}_{ij}}, \\
\end{array}
\end{multline}\]</span>
<p>where <span class="math inline">\({d}_{ij}\)</span> denotes the phylogenetic distance between the tips <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>. The above property proves useful when there is a prior knowledge that the observed population is at equilibrium, because one can use the trait variance in the population, <span class="math inline">\(\sigma_z^2=\sigma_{\infty}^2+\sigma_e^2\)</span> as model parameter. The corresponding parametrization is:</p>
<span class="math display">\[\begin{align}
&lt;g_0, \alpha,\theta,\sigma_z^2,\sigma_e&gt; &amp; \to&lt;g_0, \alpha,\theta,\sigma=\sqrt{2\alpha(\sigma_{z}^2-\sigma_e^2)},\sigma_e&gt;
\end{align}\]</span>
<p>With this parametrization, one can specify an informed prior for <span class="math inline">\(\sigma_z^2\)</span> based on empirical estimates on similar data.</p>
<p>Another important aspect of the above property is that it helps to better understand the selection stregnth parameter <span class="math inline">\(\alpha\)</span>. As it turns out, <span class="math inline">\(\alpha\)</span> can have two different biological interpretations. On the one hand, <span class="math inline">\(\alpha\)</span> defines the rate of convergence of the population mean towards the long-term optimum <span class="math inline">\(\theta\)</span>. This rate is bigger for bigger values of <span class="math inline">\(\alpha\)</span> and for bigger deviations from <span class="math inline">\(\theta\)</span>. Thus, <span class="math inline">\(\alpha\)</span> is considered as selection strength or rate of adaptation under stabilizing selection. On the other hand, it is possible to assume that the majority of the tips and their mrca’s are far enough from the root, so that <span class="math inline">\(\Sigma_{ij}\)</span> can be viewed as an exponentially decreasing function of the phylogenetic distance <span class="math inline">\({d}_{ij}\)</span>. Seen from that angle, the parameter <span class="math inline">\(\alpha\)</span> can be interpreted as the rate of phenotypic decorrelation between tips, due to genetic drift. When interpreting the results of a model fit, it is important to be aware of this dual interpretation of <span class="math inline">\(\alpha\)</span>. In many cases (e.g. in ultrametric macro-evolutionary tree), the only source of information for inferring <span class="math inline">\(\alpha\)</span> are the observed differences between the tips in the tree. Thus, in the absence of additional evidence, it can be erronous to assume that the inferred value of <span class="math inline">\(\alpha\)</span> informs stabilizing selection and an adaptation rate towards <span class="math inline">\(\theta\)</span>.</p>
<p>A likelihood ratio test between the ML POUMM and PMM fits can be used to test if the inferred parameter <span class="math inline">\(\alpha\)</span> is significantly above 0. As pointed out in the previous paragraph, a significantly positive <span class="math inline">\(\alpha\)</span> does not necessarily imply stabilizing selection towards <span class="math inline">\(\theta\)</span>. Further, it is important to note that the value of <span class="math inline">\(\alpha\)</span> can only be interpreted with respect to the time scale of the phylogeny. It can be more intuitive to consider the phylogenetic half-life, <span class="math inline">\(t_{1/2}=\frac{\ln(2)}{\alpha}\)</span>, which equals the time it takes for a species entering a new niche to evolve halfway toward its new expected optimum <span class="citation">(Hansen 1997)</span>.</p>
</div>
<div id="phylogenetic-heritability" class="section level2">
<h2 class="hasAnchor">
<a href="#phylogenetic-heritability" class="anchor"></a>Phylogenetic heritability</h2>
<p>The term <em>phylogenetic heritability</em>, introduced with the phylogenetic mixed model (PMM) <span class="citation">(Housworth, Martins, and Lynch 2004)</span>, measures the proportion of phenotypic variance in a population attributable to heritable factors, such as genes, as opposed to non-heritable factors, such as environment and measurement error. Although this concept has been applied mostly in the context of the original PMM, i.e. under the assumption of Brownian motion, the same concept applies to any evolutionary model allowing for the estimation of measurement error (ME) <span class="citation">(Hansen and Bartoszek 2012)</span>. The <em>phylogenetic heritability</em> is defined as the expected proportion of phenotypic variance attributable to <span class="math inline">\(g\)</span> at the tips of the tree, <span class="math inline">\(\sigma^2(g)/\left[\sigma^2(g)+\sigma_e^2\right]\)</span> <span class="citation">(Housworth, Martins, and Lynch 2004)</span>. This definition is a phylogenetic variant of the definition of broad-sense heritability, <span class="math inline">\(H^2\)</span>, from quantitative genetics <span class="citation">(Lynch and Walsh 1998)</span>. However, in the case of a trait evolving along a phylogeny, the expected genotypic variance, <span class="math inline">\(\sigma^2(g)\)</span>, and, therefore, the phylogenetic heritability, are functions of time. Depending on the applicaiton, the following three types of phylogenetic heritability might all be of interest:</p>
<ul>
<li>Expectation at the mean root-tip distance (<span class="math inline">\(\bar{t}\)</span>): <span class="math inline">\(H_{\bar{t}}^2:=\left[\sigma^2\, \frac{\left(1-e^{-2\alpha \bar{t}}\right)}{2\alpha}\right]/\left[\sigma^2\, \frac{\left(1-e^{-2\alpha \bar{t}}\right)}{2\alpha}+\sigma_e^2\right]\)</span>. This definition gives rize to three parametrizations where <span class="math inline">\(H_{\bar{t}}^2\)</span> is a free model parameter, while one of the parameters <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\sigma\)</span> or <span class="math inline">\(\sigma_e\)</span> is a function of <span class="math inline">\(H_{\bar{t}}^2\)</span> and the two other parameters. These parametrizations can be expressed as:
<span class="math display">\[\begin{align}
&lt;g_0, \alpha, \theta, H_{\bar{t}}^2, \sigma_e&gt; &amp; \to&lt;g_0, \alpha, \theta, \sigma=\sqrt{\frac{2\alpha H_{\bar{t}}^2 \sigma_e^2}{(1-e^{-2\alpha t})(1-H_{\bar{t}}^2)}}, \sigma_e&gt;\label{eq:sigmaH2alphasigmae} \\
&lt;g_0, H_{\bar{t}}^2, \theta, \sigma, \sigma_e&gt; &amp; \to&lt;g_0, \alpha=\frac{y+W(y e^{-y})}{2\bar{t}}, \theta, \sigma, \sigma_e&gt;\label{eq:alphaH2sigmasigmae} \\
&lt;g_0, \alpha, \theta, \sigma, H_{\bar{t}}^2&gt; &amp; \to&lt;g_0, \alpha, \theta, \sigma, \sigma_e=\sqrt{\frac{\sigma^2(1-e^{-2\alpha \bar{t}})}{2 \alpha}(\frac{1}{H_{\bar{t}}^2} - 1)}&gt;,\label{eq:sigmaeH2alphasigma}
\end{align}\]</span>
<p>where <span class="math inline">\(y=\frac{\bar{t}\sigma^2}{\sigma_e^2 H_{\bar{t}}^2} - \frac{\bar{t}\sigma^2}{\sigma_e^2}\)</span> and <span class="math inline">\(W\)</span> is the Lambert-W function.</p>
</li>
<li>Expectation at equilibrium of the OU-process (<span class="math inline">\(t\to\infty\)</span>): <span class="math inline">\(H_{\infty}^2:=\lim_{\bar{t}\to\infty}H_{\bar{t}}^2=\frac{\sigma^2}{\sigma^2+2\alpha\sigma_e^2}\)</span>. By taking the limit <span class="math inline">\(\bar{t}\to\infty\)</span> in equations ,  and , we obtain the corresponding parametrizations:
<span class="math display">\[\begin{align}
&lt;g_0, \alpha, \theta, H_{\infty}^2, \sigma_e&gt; &amp; \to&lt;g_0, \alpha, \theta, \sigma=\sqrt{\frac{2\alpha H_{\infty}^2 \sigma_e^2}{1-H_{\infty}^2}}, \sigma_e&gt;\label{eq:sigmaH2alphasigmaetInf} \\
&lt;g_0, H_{\infty}^2, \theta, \sigma, \sigma_e&gt; &amp; \to&lt;g_0, \alpha=\frac{\sigma^2 (1-H_{\infty}^2)}{2H_{\infty}^2\sigma_e^2}, \theta, \sigma, \sigma_e&gt;\label{eq:alphaH2sigmasigmaetInf} \\
&lt;g_0, \alpha, \theta, \sigma, H_{\infty}^2&gt; &amp; \to&lt;g_0, \alpha, \theta, \sigma, \sigma_e=\sqrt{\frac{\sigma^2}{2 \alpha}(\frac{1}{H_{\infty}^2} - 1)}&gt;,\label{eq:sigmaeH2alphasigmatInf}
\end{align}\]</span>
</li>
<li>Empirical (time-independent) version of the heritability based on the sample phenotypic variance <span class="math inline">\(s^2(\textbf{z})\)</span> : <span class="math inline">\(H_e^2:=1-\sigma_e^2/s^2(\textbf{z})\)</span>. This definition is useful when the tree is non-ultrametric but there is sufficient evidence that the empirical distribution of the trait is stationary along the tree. In this case, <span class="math inline">\(s^2(\textbf{z})\)</span> coincides with the sum of the OU variance at equilibrium and <span class="math inline">\(\sigma_e^2\)</span>. The corresponding parametrization is:
<span class="math display">\[\begin{align}
&lt;g_0, \alpha, \theta, \sigma, H_{e}^2&gt; &amp; \to&lt;g_0, \alpha, \theta, \sigma, \sigma_e=\sqrt{s^2(\textbf{z})(1-H_e^2)}&gt;
\label{eq:sigmaeH2e}
\end{align}\]</span>
</li>
</ul>
<p>To learn about different parametrizations supported by the POUMM package, read the documentation page <code><a href="../reference/specPOUMM.html">?specifyPOUMM</a></code>.</p>
</div>
</div>
<div id="using-the-poumm-fit" class="section level1">
<h1 class="hasAnchor">
<a href="#using-the-poumm-fit" class="anchor"></a>Using the POUMM fit</h1>
<p>Whenever presented with data consisting of a rooted phylogenetic tree with observed trait-values at its tips, the <strong>POUMM</strong> package can be used to answer THE following questions about the population distribution, the evolution and the phylogenetic signal of the trait in question.</p>
<ul>
<li>Is the POUMM an appropriate model for the data?</li>
<li>Assuming that the trait undergoes stabilizing selection, what are the long-term optimum value and the rate of convergence towards it?</li>
<li>What is the long-term equilibrium distribution of the trait values in the population?</li>
<li>To what extent are the observed trait-values determined by heritable (i.e. genetic) versus non-heritable (i.e. environmental) factors?</li>
</ul>
<!--
TODO
* What is the profile of gradual loss of phenotypic resemblance between pair-tips as a function of their phylogenetic distance?
* What are the maximum-likelihood genotypic values of the root and tips of the tree?
--><div id="is-the-poumm-an-appropriate-model-for-the-data" class="section level2">
<h2 class="hasAnchor">
<a href="#is-the-poumm-an-appropriate-model-for-the-data" class="anchor"></a>Is the POUMM an appropriate model for the data?</h2>
<p>The first step to answering that question is to visualize the data and check for obvious violations of the POUMM assumptions. The POUMM method expects that the trait-values at the tips are a sample from a multivariate normal distribution. With an ultrametric species tree, where all tips are equally distant from the root, this assumption translates in having all trait-values be realizations of identically distributed normal random variables. In the case of a non-ultrametric tree, it is far more useful to look at a sequence of box-whisker or violin plots of the trait-values, gouped by their root-tip distance.</p>
<p>Once visualizing the data has confirmed its normality, we recommend comaparing the POUMM-fit with a fit from a NULL-model such as the phylogenetic mixed model (PMM) <span class="citation">(Housworth, Martins, and Lynch 2004)</span>. Since the PMM is nested in the POUMM, i.e. in the limit <span class="math inline">\(\alpha\to0\)</span>, the POUMM model is equivalent to a PMM model with the same initial genotypic value <span class="math inline">\(g_0\)</span> and unit-time variance <span class="math inline">\(\sigma\)</span>, it is easy to fit a PMM model to the data by fixing the value of the parameter <span class="math inline">\(\alpha\)</span> to 0:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">specPMM &lt;-<span class="st"> </span><span class="kw"><a href="../reference/specPOUMM.html">specifyPMM</a></span>(z[<span class="dv">1</span><span class="op">:</span>N], tree)
fitPMM &lt;-<span class="st"> </span><span class="kw"><a href="../reference/POUMM.html">POUMM</a></span>(z[<span class="dv">1</span><span class="op">:</span>N], tree, <span class="dt">spec =</span> specPMM, <span class="dt">doMCMC=</span><span class="ot">FALSE</span>)</code></pre></div>
<p>Now a likelihood-ratio test between the maximum likelihood fits clearly shows that the POUMM significantly outperforms the PMM to the data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lmtest<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/lmtest/topics/lrtest">lrtest</a></span>(fitPMM, fitPOUMM2)</code></pre></div>
<pre><code>## Likelihood ratio test
## 
## Model 1: fitPMM
## Model 2: fitPOUMM2
##   #Df LogLik Df Chisq Pr(&gt;Chisq)    
## 1   3 -112.7                        
## 2   5  -53.3  2   119     &lt;2e-16 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
<p>Since <code>lrtest</code> only uses the ML-fit, to save time, we disabled the MCMC fit by specifying <code>doMCMC = FALSE</code>. In real situations, though, it is always recommended to enable the MCMC fit, since it can improve the ML-fit if it finds a region of higher likelihood in the parameter space that has not been discovered by the ML-fit.</p>
<p>As an exersise, we can generate data under the PMM model and see if a POUMM fit on that data remains significantly better than a PMM fit:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gBM &lt;-<span class="st"> </span><span class="kw"><a href="../reference/rVNodesGivenTreePOUMM.html">rVNodesGivenTreePOUMM</a></span>(tree, g0, <span class="dt">alpha =</span> <span class="dv">0</span>, <span class="dt">theta =</span> <span class="dv">0</span>, <span class="dt">sigma =</span> sigma)
zBM &lt;-<span class="st"> </span>gBM <span class="op">+</span><span class="st"> </span>e

fitPMM_on_zBM &lt;-<span class="st"> </span><span class="kw"><a href="../reference/POUMM.html">POUMM</a></span>(zBM[<span class="dv">1</span><span class="op">:</span>N], tree, <span class="dt">spec =</span> specPMM, <span class="dt">doMCMC =</span> <span class="ot">FALSE</span>)
fitPOUMM_on_zBM &lt;-<span class="st"> </span><span class="kw"><a href="../reference/POUMM.html">POUMM</a></span>(zBM[<span class="dv">1</span><span class="op">:</span>N], tree, <span class="dt">doMCMC =</span> <span class="ot">FALSE</span>)

lmtest<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/lmtest/topics/lrtest">lrtest</a></span>(fitPMM_on_zBM, fitPOUMM_on_zBM)</code></pre></div>
<pre><code>## Likelihood ratio test
## 
## Model 1: fitPMM_on_zBM
## Model 2: fitPOUMM_on_zBM
##   #Df LogLik Df Chisq Pr(&gt;Chisq)    
## 1   3   -131                        
## 2   5   -225  2   187     &lt;2e-16 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div id="assuming-that-the-trait-undergoes-stabilizing-selection-what-are-the-long-term-optimum-value-and-the-rate-of-convergence-towards-it" class="section level2">
<h2 class="hasAnchor">
<a href="#assuming-that-the-trait-undergoes-stabilizing-selection-what-are-the-long-term-optimum-value-and-the-rate-of-convergence-towards-it" class="anchor"></a>Assuming that the trait undergoes stabilizing selection, what are the long-term optimum value and the rate of convergence towards it?</h2>
<p>To answer this question, consider the estimated values of the POUMM-parameters <span class="math inline">\(\theta\)</span> and <span class="math inline">\(\alpha\)</span>. Note that the parameter <span class="math inline">\(\theta\)</span> is relevant only if the value of the parameter <span class="math inline">\(\alpha\)</span> is significantly positive. One could accept that the ML-estimate for <span class="math inline">\(\alpha\)</span> is significantly positive if a likelihood ratio test between a ML PMM and POUMM fits gives a p-value below a critical level (see the question above for an example). An inisignificant value of <span class="math inline">\(\alpha\)</span> reveals that the hypothesis of neutral drift (Brownian motion) cannot be rejected.</p>
</div>
<div id="what-is-the-long-term-equilibrium-normal-distribution-of-the-trait-values-in-the-population" class="section level2">
<h2 class="hasAnchor">
<a href="#what-is-the-long-term-equilibrium-normal-distribution-of-the-trait-values-in-the-population" class="anchor"></a>What is the long-term equilibrium normal distribution of the trait values in the population?</h2>
<p>The long-term mean is given by the estimated parameter <code>theta</code>. The long-term variance is given by the sum <code>sigmaG2tInf+sigmae^2</code> taken from the summary table of a POUMM-fit.</p>
</div>
<div id="to-what-extent-are-the-observed-trait-values-determined-by-heritable-i-e-genetic-versus-non-heritable-i-e-environmental-factors" class="section level2">
<h2 class="hasAnchor">
<a href="#to-what-extent-are-the-observed-trait-values-determined-by-heritable-i-e-genetic-versus-non-heritable-i-e-environmental-factors" class="anchor"></a>To what extent are the observed trait-values determined by heritable (i.e. genetic) versus non-heritable (i.e. environmental) factors?</h2>
<p>In other words, what is the proportion of observable phenotypic variance attributable to the phylogeny? To answer this question, the POUMM package allows to estimate the phylogenetic heritability of the trait. Assuming that the tree represents the genetic relationship between individuals in a population, <span class="math inline">\(H_\bar{t}^2\)</span> provides an estimate for the broad-sense heritability <span class="math inline">\(H^2\)</span> of the trait in the population.</p>
<p>When the goal is to estimate <span class="math inline">\(H_{\bar{t}}^2\)</span> (<code>H2tMean</code>), it is imortant to specify an uninformed prior for it. Looking at the densities for chain 1 (red) on the previous figures, it becomes clear that the default prior favors values of <code>H2tMean</code>, which are either close to 0 or close to 1. Since by definition <span class="math inline">\(H_{\bar{t}}^2\in[0,1]\)</span>, a reasonable uninformed prior for it is the standard uniform distribution. We set this prior by using the <code>specifyPOUMM_ATH2tMeanSeG0</code> function. This specifies that the POUMM fit should be done on a parametrization <span class="math inline">\(&lt;\alpha,\theta,H_{\bar{t}}^2,\sigma_e,g_0&gt;\)</span> rather than the standard parametrization <span class="math inline">\(&lt;\alpha,\theta,\sigma,\sigma_e,g_0&gt;\)</span>. It also specifies a uniform prior for <span class="math inline">\(H_{\bar{t}}^2\)</span>. You can explore the members of the specification list to see the different settings:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">specH2tMean &lt;-<span class="st"> </span><span class="kw"><a href="../reference/specPOUMM.html">specifyPOUMM_ATH2tMeanSeG0</a></span>(z[<span class="dv">1</span><span class="op">:</span>N], tree, <span class="dt">nSamplesMCMC =</span> <span class="fl">4e5</span>)
<span class="co"># Mapping from the sampled parameters to the standard POUMM parameters:</span>
specH2tMean<span class="op">$</span>parMapping</code></pre></div>
<pre><code>## function (par) 
## {
##     if (is.matrix(par)) {
##         par[, 3] &lt;- sigmaOU(par[, 3], par[, 1], par[, 4], tMean)
##         colnames(par) &lt;- c("alpha", "theta", "sigma", "sigmae", 
##             "g0")
##     }
##     else {
##         par[3] &lt;- sigmaOU(par[3], par[1], par[4], tMean)
##         names(par) &lt;- c("alpha", "theta", "sigma", "sigmae", 
##             "g0")
##     }
##     par
## }
## &lt;bytecode: 0x7faa6c531df0&gt;
## &lt;environment: 0x7faa6c546868&gt;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Prior for the MCMC sampling</span>
specH2tMean<span class="op">$</span>parPriorMCMC</code></pre></div>
<pre><code>## function (par) 
## {
##     dexp(par[1], rate = tMean/6.931, log = TRUE) + dnorm(par[2], 
##         zMean, 2 * zSD, TRUE) + dunif(par[3], min = 0, max = 1, 
##         log = TRUE) + dexp(par[4], rate = 2/zSD, log = TRUE) + 
##         dnorm(par[5], zMean, 2 * zSD, log = TRUE)
## }
## &lt;bytecode: 0x7faa6c53b3f8&gt;
## &lt;environment: 0x7faa6c546868&gt;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Bounds for the maximum likelihood search</span>
specH2tMean<span class="op">$</span>parLower</code></pre></div>
<pre><code>##   alpha   theta H2tMean  sigmae      g0 
##  0.0000 -4.6686  0.0000  0.0000 -0.3944</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">specH2tMean<span class="op">$</span>parUpper</code></pre></div>
<pre><code>##   alpha   theta H2tMean  sigmae      g0 
## 14.0440  7.7297  0.9900  0.6851  3.4555</code></pre>
<p>Then we fit the model:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fitH2tMean &lt;-<span class="st"> </span><span class="kw"><a href="../reference/POUMM.html">POUMM</a></span>(z[<span class="dv">1</span><span class="op">:</span>N], tree, <span class="dt">spec =</span> specH2tMean)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(fitH2tMean, <span class="dt">stat =</span> <span class="kw">c</span>(<span class="st">"H2tMean"</span>, <span class="st">"H2e"</span>, <span class="st">"H2tInf"</span>, <span class="st">"sigmae"</span>), 
     <span class="dt">showUnivarDensityOnDiag =</span> <span class="ot">TRUE</span>, 
     <span class="dt">doZoomIn =</span> <span class="ot">TRUE</span>, <span class="dt">doPlot =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p><img src="InterpretingPOUMM_files/figure-html/unnamed-chunk-7-1.png" width="691.2"><img src="InterpretingPOUMM_files/figure-html/unnamed-chunk-7-2.png" width="691.2"><img src="InterpretingPOUMM_files/figure-html/unnamed-chunk-7-3.png" width="691.2"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(fitH2tMean)[stat <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">"H2tMean"</span>, <span class="st">"H2e"</span>, <span class="st">"H2tInf"</span>, <span class="st">"sigmae"</span>)]</code></pre></div>
<pre><code>##       stat   N    MLE PostMean           HPD   ESS  G.R.
## 1: H2tMean 500 0.4991   0.5024 0.3214,0.6865 644.7 1.001
## 2:  sigmae 500 0.2052   0.2058 0.1721,0.2373 720.0 1.002
## 3:     H2e 500 0.6411   0.6365 0.5260,0.7528 720.0 1.002
## 4:  H2tInf 500 0.5015   0.5065 0.3198,0.6848 646.2 1.001</code></pre>
<p>Now we see that the prior density for <code>H2tMean</code> is nearly uniform. It becomes clear that the process has converged to its long-term heritability since the intervals for <code>H2tMean</code> and <code>H2tInf</code> are nearly the same. Notice, though, that the estimate for the empirical heritability <code>H2e</code> is shifted towards 1 compared to <code>H2tMean</code> and <code>H2tInf</code>. This shows an important difference between <code>H2e</code> and the time-dependent formulae for phylogenetic heritability: <code>H2e</code> takes into account all values of z including those at the very beginning when the process was far away from equilibrium. Thus the estimated phenotypic variance over all trait-values at all times can be substantially bigger compared to the current trait-variance in the population:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Compare global empirical heritability</span>
H2eGlobal &lt;-<span class="st"> </span><span class="kw"><a href="../reference/PhylogeneticH2.html">H2e</a></span>(z[<span class="dv">1</span><span class="op">:</span>N], <span class="dt">sigmae =</span> <span class="kw">coef</span>(fitH2tMean)[<span class="st">'sigmae'</span>])
<span class="co"># versus recent empirical heritability</span>
H2eRecent &lt;-<span class="st"> </span><span class="kw"><a href="../reference/PhylogeneticH2.html">H2e</a></span>(z[<span class="dv">1</span><span class="op">:</span>N], tree, <span class="dt">sigmae =</span> <span class="kw">coef</span>(fitH2tMean)[<span class="st">'sigmae'</span>], <span class="dt">tFrom =</span> <span class="dv">5</span>)
<span class="kw">print</span>(<span class="kw">c</span>(H2eGlobal, H2eRecent))</code></pre></div>
<pre><code>## [1] 0.6411 0.5010</code></pre>
<p>To learn more about different ways to specify the POUMM fit, read the documentation page <code><a href="../reference/specPOUMM.html">?specifyPOUMM_ATH2tMeanSeG0</a></code>. <!--
TODO
## What is the profile of gradual loss of phenotypic resemblance between pair-tips as a function of their phylogenetic distance?


## What are the maximum-likelihood genotypic values of the root and tips of the tree?
--></p>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Butler:2004ce">
<p>Butler, M A, and A A King. 2004. “Phylogenetic comparative analysis: A modeling approach for adaptive evolution.” <em>American Naturalist</em> 164 (6): 683–95.</p>
</div>
<div id="ref-Hansen:1997ek">
<p>Hansen, Thomas F. 1997. “Stabilizing Selection and the Comparative Analysis of Adaptation.” <em>Evolution; International Journal of Organic Evolution</em> 51 (5): 1341–51.</p>
</div>
<div id="ref-Hansen:2012et">
<p>Hansen, Thomas F, and Krzysztof Bartoszek. 2012. “Interpreting the evolutionary regression: the interplay between observational and biological errors in phylogenetic comparative studies.” <em>Systematic Biology</em> 61 (3): 413–25.</p>
</div>
<div id="ref-Harmon:2010gh">
<p>Harmon, Luke J, Jonathan B Losos, T Jonathan Davies, Rosemary G Gillespie, John L Gittleman, W Bryan Jennings, Kenneth H Kozak, et al. 2010. “Early bursts of body size and shape evolution are rare in comparative data.” <em>Evolution; International Journal of Organic Evolution</em> 64 (8): 2385–96.</p>
</div>
<div id="ref-Hodcroft:2014es">
<p>Hodcroft, Emma, Jarrod D Hadfield, Esther Fearnhill, Andrew Phillips, David Dunn, Siobhan O’Shea, Deenan Pillay, Andrew J Leigh Brown, on behalf of the UK HIV Drug Resistance Database, and the UK CHIC Study. 2014. “The Contribution of Viral Genotype to Plasma Viral Set-Point in HIV Infection.” <em>PLoS Pathogens</em> 10 (5): e1004112.</p>
</div>
<div id="ref-Housworth:2004gj">
<p>Housworth, Elizabeth A, Emília P Martins, and Michael Lynch. 2004. “The phylogenetic mixed model.” <em>The American Naturalist</em> 163 (1): 84–96.</p>
</div>
<div id="ref-Lynch:1991db">
<p>Lynch, Michael. 1991. “Methods for the Analysis of Comparative Data in Evolutionary Biology.” <em>Evolution; International Journal of Organic Evolution</em> 45 (5): 1065–80.</p>
</div>
<div id="ref-Lynch:1998vx">
<p>Lynch, Michael, and Bruce Walsh. 1998. <em>Genetics and Analysis of Quantitative Traits</em>. Sinauer Associates Incorporated.</p>
</div>
<div id="ref-Mitov:2016kd">
<p>Mitov, Venelin, and Tanja Stadler. 2016. “The heritability of pathogen traits - definitions and estimators.” <em>Unpublished Data</em> https://www.biorxiv.org/content/early/2016/06/12/058503, last accessed June 12, 2016 (June).</p>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#parametrizing-the-poumm">Parametrizing the POUMM</a><ul class="nav nav-pills nav-stacked">
<li><a href="#trait-distribution-at-equilibrium">Trait distribution at equilibrium</a></li>
      <li><a href="#phylogenetic-heritability">Phylogenetic heritability</a></li>
      </ul>
</li>
      <li>
<a href="#using-the-poumm-fit">Using the POUMM fit</a><ul class="nav nav-pills nav-stacked">
<li><a href="#is-the-poumm-an-appropriate-model-for-the-data">Is the POUMM an appropriate model for the data?</a></li>
      <li><a href="#assuming-that-the-trait-undergoes-stabilizing-selection-what-are-the-long-term-optimum-value-and-the-rate-of-convergence-towards-it">Assuming that the trait undergoes stabilizing selection, what are the long-term optimum value and the rate of convergence towards it?</a></li>
      <li><a href="#what-is-the-long-term-equilibrium-normal-distribution-of-the-trait-values-in-the-population">What is the long-term equilibrium normal distribution of the trait values in the population?</a></li>
      <li><a href="#to-what-extent-are-the-observed-trait-values-determined-by-heritable-i-e-genetic-versus-non-heritable-i-e-environmental-factors">To what extent are the observed trait-values determined by heritable (i.e. genetic) versus non-heritable (i.e. environmental) factors?</a></li>
      </ul>
</li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Venelin Mitov.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
